imagelab/ImgProvider.java

```diff
@@ -1,502 +1,610 @@
 package imagelab;
 
-import imagelab.*;
-import sound.*;
-import java.awt.*;
-import java.awt.image.*;
-import javax.imageio.*;
-import java.util.Iterator;
-import javax.swing.*;
-import java.io.*;
+import sound.Chord;
+import sound.Music;
+import sound.Note;
+import sound.Scale;
+import sound.Tune;
+
+import javax.imageio.ImageIO;
+import javax.swing.JComponent;
+import javax.swing.JFileChooser;
+import javax.swing.JOptionPane;
+import javax.swing.JFrame;
+import java.awt.Image;
+import java.awt.image.BufferedImage;
+import java.awt.image.MemoryImageSource;
+import java.awt.image.PixelGrabber;
+import java.io.File;
+import java.io.IOException;
+
 /**
  * ImgProvider is responsible for managing a single image
  * (loading, filtration, rendering, etc.).
+ *
  * @author Dr. Aaron Gordon
  * @author Dr. Jody Paul
  * @version 1.1
  */
 public class ImgProvider extends JComponent {
     /** Serialization version. */
     private static final long serialVersionUID = 11L;
-    static Thread playThread;
-    static boolean  all;
-    /** true if this ImgProvider currently holds an image; false otherwise. */
-    boolean         isLoaded;
+    /** Thread that allows asynchronus activity. */
+    private static Thread playThread;
+    /** Debug variable to show all filters. */
+    private static boolean all;
+    /** True if this ImgProvider currently holds an image; false otherwise. */
+    private boolean isLoaded;
     /** Image height in pixels. */
-    int             pixheight;
+    protected int pixheight;
     /** Image width in pixels. */
-    int             pixwidth;
+    protected int pixwidth;
     /** The raw image. */
-    Image           img;
+    private Image img;
     /** Holders for the color and alpha components of the image. */
-    short [][]      red, green, blue, alpha;
+    private short[][] green, red, blue, alpha;
     /** To retrieve pixels from the image. */
-    PixelGrabber    grab;
+    private PixelGrabber grab;
     /** Holder for the pixels from the image. */
-    int []          pix;
+    protected int[] pix;
     /** X-axis increment used for trimming the image. */
-    protected int   xinc= 0;
+    private int xinc = 0;
     /** Y-axis increment used for trimming the image. */
-    protected int   yinc = 0;
+    private int yinc = 0;
     /** Holder for the filename of the file that contains the image. */
-    protected String imgName;
+    private String imgName;
     /** Used for assigning unique IDs to ImgProviders. Incremented when used. */
-    protected static int count = 0;
+    private static int count = 0;
```

Comment:
As far as we know, there is no reason for these variables to be accessed outside of the class. Also, these variables were changed because they were showing checkstyle errors.

Reply:
> As far as we know, there is no reason for these variables to be accessed outside of the class.

Then it makes sense to me for them all to have `private` visibility and there be no accessor or mutator methods.